### PR TITLE
Completed the app

### PR DESCRIPTION
**Pull request by puntnomads**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -42,9 +60,41 @@ def confirmation(order_id):
...
-    # TODO other stuff has to be calculated here.
-    return render_template('confirmation.jinja', order_id=order_id, title='Order Confirmation')
+        return redirect(url_for('index'))
+    amount_paid = round(float(order['product_id']['amount_paid']), 2)
+    item_price = float(PRODUCTS[int(order['product_id']['id'])]['price'])
+    change_due = round(amount_paid - item_price, 2)
+    change_left = change_due * 100
+    denominations = []
+    for item in DENOMINATIONS:
+        if change_left - item['value'] * 2 >= 0:
```
Comment (tomhamiltonstubber) [inline app.py]: This doesn't work, and to be honest doesn't really make sense...

image

File: templates/orders.jinja [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{% extends 'base.jinja' %}
```
Comment (tomhamiltonstubber) [inline templates/orders.jinja]: Good to create this, but then we need a link to it:smile: And the buyers name and amount spend would be a sensible addition to someone viewing this, no?

File: app.py [hunk 3]
```diff
@@ -24,15 +24,33 @@ def record_order(product_id):
...
-    return render_template('index.jinja', products=PRODUCTS, title='Order Form', **context)
+        id = request.form['product'].strip()
+        amount_paid = request.form['paid'].strip()
+        if not id:
+            flash('Please choose a product', 'danger')
+        elif float(amount_paid) < PRODUCTS[int(id)]['price']:
+            flash('Please pay more money', 'danger')
+        else:
+            flash('Order Placed Successfully', 'success')
+            product_id = {
```
Comment (tomhamiltonstubber) [inline app.py]: Hmm small thing, but this isn't an id.

File: app.py [hunk 4]
```diff
@@ -24,15 +24,33 @@ def record_order(product_id):
...
 
 
 @app.route('/', methods=['POST', 'GET'])
 def index():
     context = {}
     if request.method == 'POST':
-        flash('Order Placed Successfully', 'success')
-        # TODO
-    return render_template('index.jinja', products=PRODUCTS, title='Order Form', **context)
+        id = request.form['product'].strip()
```
[the remark this answers is not in the document]
Comment (tomhamiltonstubber) [reply app.py]: Also do you need to strip it? You can just call `int(request.form['product'])`

**Comment by tomhamiltonstubber** [comment]
Okay I've given you quite a bit to work on, have another try and we'll see how it's going.

**Comment by puntnomads** [comment]
What am I supposed to be working on? I think I have completed the objectives in the README.

File: app.py [hunk 5]
```diff
@@ -42,9 +60,41 @@ def confirmation(order_id):
 
     order = orders.get(order_id)
     if order is None:
-        pass  # TODO what do we do here?
-    # TODO other stuff has to be calculated here.
-    return render_template('confirmation.jinja', order_id=order_id, title='Order Confirmation')
+        return redirect(url_for('index'))
+    amount_paid = round(float(order['product_id']['amount_paid']), 2)
+    item_price = float(PRODUCTS[int(order['product_id']['id'])]['price'])
```
[the remark this answers is not in the document]
Comment (puntnomads) [reply app.py]: I don't know what this means.

**Pull request closed without merging**